crates/ruff_graph/src/resolver.rs

```diff
@@ -0,0 +1,102 @@
+use crate::db::ModuleDb;
+use red_knot_python_semantic::{Module, ModuleName};
+use ruff_db::files::FilePath;
+
+use crate::collector::CollectedImport;
+use ruff_python_ast::name::{QualifiedName, QualifiedNameBuilder};
+use ruff_python_stdlib::identifiers::is_identifier;
+use std::borrow::Cow;
+
+/// Collect all imports for a given Python file.
+pub(crate) struct Resolver<'ast> {
+    module_path: Option<&'ast [String]>,
+    db: &'ast ModuleDb,
+}
```

Comment:
Nit: the use of the `ast` lifetime for `db` is somewhat confusing. Maybe just use `'a`.